codespell: workflow, config + some (quite a few) typos fixed

yarikoptic opened the pull request:
Probably the most  boring PR to review ;)

Individual commits might be easier to digest

yarikoptic committed:
Add rudimentary codespell config

yarikoptic committed:
Add github action to codespell master on push and PRs

yarikoptic committed:
Add github action to codespell master on push and PRs

yarikoptic committed:
Add rudimentary codespell config

yarikoptic committed:
Various skips

yarikoptic committed:
Various skips

yarikoptic committed:
Fix some ambigous typos and ignore one more one done on purpose

yarikoptic committed:
Fix some ambigous typos and ignore one more one done on purpose

yarikoptic committed:
Some more ignores

yarikoptic committed:
Some more ignores

yarikoptic committed:
[DATALAD RUNCMD] run codespell throughout

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

baskaryan committed:
merge

baskaryan commented:
this is awesome @yarikoptic! looks like the new codespell CI check is failing :') could we add a make command or script to make it easy to run codespell locally?

yarikoptic committed:
ignore more files - css and min.js added recently causing outdated PR to fail

also ignored example_data since there are abbreviated words etc in .xml

yarikoptic committed:
[DATALAD RUNCMD] run codespell throughout

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

baskaryan committed:
Merge branch 'master' into enh-codespell

baskaryan committed:
latest

baskaryan committed:
pull

baskaryan committed:
cr

baskaryan commented:
> right? ;) the point is that running `codespell` is sufficient if you have it installed "properly" (given that config is in pyproject.toml you need `tomli` installed too).

haha fair, but i think probably still helpful to have simple make command for other contributors. added to Makefile and contributor guidelines, if that looks ok to you happy to land!

yarikoptic commented on .github/CONTRIBUTING.md:
```diff
@@ -123,6 +123,31 @@ This can be very helpful when you've made changes to only certain parts of the p
 
+### Spellcheck
+
+Spellchecking for this project is done via [codespell](https://github.com/codespell-project/codespell).
```
note that `codespell` is not really a full blown spellchecker - it is rather a "finder of common typos", so many misspelled words could exist, and on contrary -- `codespell` could alert about some correct, but rarely used words, which are more typically appearing as typos.  So may be make it explicit here:

```suggestion
Spellchecking for this project is done via [codespell](https://github.com/codespell-project/codespell).
Note that `codespell` finds common typos, so could have false-positive (correctly spelled but rarely used) and false-negatives (not finding misspelled) words.
```

yarikoptic commented on tests/unit_tests/tools/openapi/test_specs/zapier/apispec.json:
```diff
@@ -3,7 +3,7 @@
    "info": {
       "title": "Zapier Natural Language Actions (NLA) API (Dynamic) - Beta",
       "version": "1.0.0",
-      "description": "<img src=\"https://cdn.zappy.app/945f9bf9e44126873952ec5113949c3f.png\" width=\"100\" />\n\n## Hello, friend!\nWelcome to the **Zapier Natural Language Actions API docs**. You are currently viewing the **dynamic** API.\n\nThe endpoints below are dynamically generated based on your [current user session](/login/zapier/) and [enabled actions](/demo/).\n\nThese *dynamic* endpoi ...
```
oh -- should may be this file be skipped?

baskaryan replied:
good point! will skip test data files

baskaryan committed:
Update .github/CONTRIBUTING.md

Co-authored-by: Yaroslav Halchenko <debian@onerussian.com>

baskaryan committed:
cr

baskaryan committed:
Merge branch 'enh-codespell' of github.com:yarikoptic/langchain into enh-codespell

baskaryan committed:
merge

yarikoptic commented:
Please let me know if you want me to do anything (e.g. to rebase/reapply codespell etc).

baskaryan commented:
> Please let me know if you want me to do anything (e.g. to rebase/reapply codespell etc).

all lgtm, thanks @yarikoptic!!
